casper/src/main/resources/RevVault.rho

```diff
@@ -14,24 +14,51 @@
 // ----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
 new
   RevVault, rs(`rho:registry:insertSigned:ed25519`), uriOut,
-  rl(`rho:registry:lookup`), RevAddress(`rho:rev:address`), MakeMintCh, LockboxCh, EitherCh,
+  rl(`rho:registry:lookup`), RevAddress(`rho:rev:address`), MakeMintCh, AuthKeyCh, EitherCh,
+  _getIdentity,
   _makeVault,
   _newVault,
   _getOrCreate,
   _revVault,
   stdout(`rho:io:stdout`)
 in {
   rl!(`rho:id:exunyijimapk7z43g3bbr69awqdz54kyroj9q43jgu3dh567fxsftx`, *MakeMintCh) |
-  rl!(`rho:id:9dsr55z1js13x346yhhecx66ns486i3yqf6jafrd9p9hdrrbxjqmyu`, *LockboxCh) |
+  rl!(`rho:id:4njqcsc65mt8xrfx9nsz7je7oucgywgd1tj1n1gjtw3ndsophyisro`, *AuthKeyCh) |
   rl!(`rho:id:j6trahbxycumerwpr5qype7j43b7eqh8auebwsa9nn68if47gswh73`, *EitherCh) |
-  for(@(_, MakeMint) <- MakeMintCh; @(_, Lockbox) <- LockboxCh; @(_, Either) <- EitherCh) {
+  for(@(_, MakeMint) <- MakeMintCh; @(_, AuthKey) <- AuthKeyCh; @(_, Either) <- EitherCh) {
 
     new mintCh, vaultMapStore in {
       vaultMapStore!({}) |
       @MakeMint!(*mintCh) |
       for (mint <- mintCh) {
 
-        //FIXME the result of findOrCreateGenesisVault is not protected, nor is the call to it. Both should be.
+        contract RevVault(@"deployerAuthKey", ret) = {
+
+          new deployerPubKeyBytesCh in {
+            _getIdentity!(*deployerPubKeyBytesCh) |
+            for (@deployerPubKeyBytes <- deployerPubKeyBytesCh) {
+
+              new RevAddress(`rho:rev:address`), revAddrCh in {
+                RevAddress!("fromPublicKey", deployerPubKeyBytes, *revAddrCh)|
+                for (@deployerRevAddress <- revAddrCh) {
+
+                  @AuthKey!("make", (*_revVault, deployerRevAddress), *ret)
```

Comment:
Because otherwise anybody could re-make the key by just knowing the revAddress. The only way to obtain an authKey should be by calling `RevAddress.deployerAuthKey`. Using the non-leaking `_revVault` unforgrable name is the way to achieve that. Notice that the AuthKey is structured so that passing a bare (un-bundled) name into its methods doesn't cause a leak.

Reply:
> Because otherwise anybody could re-make the key by just knowing the revAddress.

But you get the revAddress through the deploy data. Therefore in order to re-make the key someone would have to know the private key for a given revAddress which sounds about right.